iceoryx2-bb/system-types/tests/path_tests.rs

```diff
@@ -72,6 +72,24 @@ mod windows {
             .unwrap();
         assert_that!(sut, eq b"fuu\\blaaaha\\blub.ma");
     }
+
+    #[test]
+    fn path_is_absolute_works() {
+        let sut = Path::new(b"D:\\bla").unwrap();
+        assert_that!(sut.is_absolute(), eq true);
+
+        let sut = Path::new(b"k:\\fuu\\bar").unwrap();
+        assert_that!(sut.is_absolute(), eq true);
+
+        let sut = Path::new(b"0:\\a\\b").unwrap();
+        assert_that!(sut.is_absolute(), eq false);
+
+        let sut = Path::new(b"").unwrap();
+        assert_that!(sut.is_absolute(), eq false);
+
+        let sut = Path::new(b"what/ever/").unwrap();
+        assert_that!(sut.is_absolute(), eq false);
+    }
```

Comment:
It gets more complicated https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#path-normalization

There are also UNC paths but those are related to network paths. Then there is `\\.\` and `\\?\`. And since that is not fun enough, there is also `\foo\bar` which is relative to the root of the current drive

Reply:
There is also a big list of more or less obscure windows reserved file and directory names: https://help.interfaceware.com/v6/windows-reserved-file-names

This is not the full list, and the list varies of course from windows to windows version.

When I read this, I decided that making `path` on windows safe in the sense that it always contains a legal non-reserved file/path name is nearly impossible. So it is best effort to cover the 99%. For the remaining stuff we have the low level errors, panics and a useful detailed log output.

Reply:
How about just using `std::path::Path` on Windows? We probably don't need Windows to be no-std

Reply:
Yes this makes sense, and we maybe shouldn't stop at `std::path::Path` and also reuse other std constructs. But I would keep this for another PR and create a good-first-issue for that.